content/en/platform/corda/5.0/developing-applications/getting-started/utxo-advanced-ledger-extensions/commands-example.md

```diff
@@ -0,0 +1,243 @@
+    void verify(UtxoLedgerTransaction transaction);
+}
+```
+
+Now that we have a command which itself can implement contract verification constraints, we can use this as the basis for the `ExampleContractCommand` class. This needs to be a class rather than an interface, because we need to be in complete control of its implementations for security.
+We achieve this by making the default constructor package private, so that only commands within the same package can extend it; for example:
+
+```kotlin
+ public class ExampleContractCommand implements VerifiableCommand {
+  ExampleContractCommand() { }
+}
+```
+
+Next, we can implement this interface as `Create`, `Update` and `Delete` commands; for example:
+
+```kotlin
+ public class Create extends ExampleContractCommand {
+    
+    @Override
+    
+    public final void verify(UtxoLedgerTransaction transaction) {
+    
+        // Verify base Create constraints
+        // Then verify additional Create constraints implemented in derived commands
+        onVerify(transaction);
+        
+    }
+    protected void onVerify(UtxoLedgerTransaction transaction) { }
+}
+
+public class Update extends ExampleContractCommand {
+
+    @Override
+    
+    public final void verify(UtxoLedgerTransaction transaction) {
+    
+        // Verify base Update constraints
+        // Then verify additional Update constraints implemented in derived commands
+        onVerify(transaction);
+        
+    }
+    protected void onVerify(UtxoLedgerTransaction transaction) { }
+}
+
+public class Delete extends ExampleContractCommand {
+
+    @Override
+    
+    public final void verify(UtxoLedgerTransaction transaction) {
+        // Verify base Delete constraints
+        // Then verify additional Delete constraints implemented in derived commands
+        onVerify(transaction);
+    }
+    protected void onVerify(UtxoLedgerTransaction transaction) { }
+}
+```
+
+{{< note >}}
+The `Create`, `Update` and `Delete` commands are not marked final. Therefore, we can extend the contract verification constraints from these points, but we cannot extend from `ExampleContractCommand`.
+{{< /note >}}
+
+## Delegated Contract Design
+
+As we have now delegated contract verification constraint logic to the commands themselves, we must also refactor the contract to support this delegation. The contract implementation in this case becomes simpler, since it is no longer responsible for defining contract verification constraints. For example:
+
+```kotlin
+public final class ExampleContract implements Contract {
+    
+      @Override
+      
+      public void verify(UtxoLedgerTransaction transaction) {
+         List<? extends ExampleContractCommand> commands = transaction
+                  .getCommands(ExampleContractCommand.class);
+                  
+          for (ExampleContractCommand command : commands) {
+              command.verify(transaction);
+          }
+      }
+    }
+```
```

Comment:
```suggestion
```kotlin
public final class ExampleContract implements Contract {
    @Override
    public void verify(UtxoLedgerTransaction transaction) {
        List<? extends ExampleContractCommand> commands = transaction
            .getCommands(ExampleContractCommand.class);
                  
        for (ExampleContractCommand command : commands) {
            command.verify(transaction);
        }
    }
}
```
```